synapse/lib/iq.py

```diff
@@ -829,3 +830,19 @@ def istufo(self, obj):
         self.len(2, obj)
         self.isinstance(obj[0], (type(None), str))
         self.isinstance(obj[1], dict)
+
+@contextlib.contextmanager
+def redirectStdin(new_stdin):
```

Comment:
lets make this guy part of the SynTest class directly and add an example of use

Reply:
lets also include examples showing bytes vs text oriented use

Reply:
+1 moving `redirectStdin` to SynTest

Reply:
ok